sdk/python/kfp/client/client.py

```diff
@@ -1104,29 +1108,39 @@ def create_run_from_pipeline_package(
                 compile time settings).
             service_account: Specifies which Kubernetes service
                 account to use for this run.
+            experiment_id: ID of the experiment to add the run to.
 
         Returns:
             ``RunPipelineResult`` object containing information about the pipeline run.
         """
 
         #TODO: Check arguments against the pipeline function
         pipeline_name = os.path.basename(pipeline_file)
-        experiment_name = experiment_name or os.environ.get(
-            KF_PIPELINES_DEFAULT_EXPERIMENT_NAME, None)
-        overridden_experiment_name = os.environ.get(
-            KF_PIPELINES_OVERRIDE_EXPERIMENT_NAME, experiment_name)
-        if overridden_experiment_name != experiment_name:
-            warnings.warn(
-                f'Changing experiment name from "{experiment_name}" to "{overridden_experiment_name}".'
-            )
-        experiment_name = overridden_experiment_name or 'Default'
+
+        if (experiment_name is not None) and (experiment_id is not None):
+            raise ValueError(
+                'You cannot specify both experiment_name and experiment_id.')
```

Comment:
nit: Better also mention this in the docstring.

Reply:
Non-ironically love nits. Resolved!